src/main/java/app/studnicki/ox/Round.java

```diff
@@ -0,0 +1,55 @@
+package app.studnicki.ox;
+
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+import java.beans.PropertyChangeSupport;
+import java.util.HashMap;
+import java.util.Map;
+
+class Round {
+  final int dimension;
+  Map<Integer, Sign> board = new HashMap<>();
+
+  private final int limit;
+  private static final int MININUM_DIMENSION = 3;
+  private final PropertyChangeSupport propertyChangeSupport = new PropertyChangeSupport(this);
+
+  void addObserver(PropertyChangeListener listener) {
+    propertyChangeSupport.addPropertyChangeListener("filledField", listener);
+  }
+
+  /**
+   * Constructor
+   *
+   * @param dimension Integer value that defines game board size
+   *                  (square: dimension x dimension)and range of id fields (0..(dimension^2 - 1)).
+   */
+  Round(int dimension) {
+    if (dimension < MININUM_DIMENSION) {
+      throw new IllegalArgumentException("Board dimension cannot be lesser than 3!");
+    }
+    this.dimension = dimension;
+    limit = dimension * dimension;
```

Comment:
Is it a Round responsibility for sure?

Reply:
It can be delegated to something like a Board class. Can be done for next pull request.